### PR TITLE
WIP: Add file writing and reading

### PR DESCRIPTION
**Pull request by MMagueta**

### CODE REVIEW

File: src/stdlib/Builtins.fs [hunk 1]
```diff
@@ -10,13 +10,14 @@ open Fable.Core
...
     abstract read : __size: int -> string
+    abstract write : string -> unit
 
 type TextIOWrapper =
     inherit IDisposable
     inherit TextIOBase
 
 [<StringEnum>]
-type OpenTextModeUpdating =
+type OpenTextMode =
```
Comment (MMagueta) [inline src/stdlib/Builtins.fs:20]: So @dbrattli here I did something ugly. I tried to get the `OpenTextMode ` union working with its different open modes but I could only get it working by adding the read, write and update to the same union context. What would you recommend to do here?

File: test/TestBuiltins.fs [hunk 2]
```diff
@@ -8,4 +8,9 @@ let ``test print works`` () =
     let result = builtins.print "Hello, world!"
     result |> equal ()
 
+[<Fact>]
+let ``test write works`` () =
```
Comment (MMagueta) [inline test/TestBuiltins.fs:12]: I still need to fix this test, don't take it for granted

File: src/stdlib/Builtins.fs (hunk 1 above)
Comment (dbrattli) [reply src/stdlib/Builtins.fs:20]: @MMagueta Yes this looks like the right thing to do 👍

**Comment by dbrattli** [comment]
The other way would be to fix the union to be:

```fsharp
[<Erase>]
type OpenTextMode =
    | OpenTextModeUpdating of OpenTextModeUpdating
    | OpenTextModeWriting of OpenTextModeWriting
    | OpenTextModeReading of OpenTextModeReading
```

then you can do:

```fsharp
    let result = builtins.``open``(StringPath "test.txt", OpenTextModeWriting OpenTextModeWriting.Write)
```

But it feels a bit verbose, so I think it's OK to collapse them as you suggest. The inspiration for the code is btw:

- https://github.com/python/typeshed/blob/master/stdlib/_typeshed/__init__.pyi#L156
- https://github.com/python/typeshed/blob/master/stdlib/io.pyi#L165

**Comment by MMagueta** [comment]
> The other way would be to fix the union to be:
> 
> ```fsharp
> [<Erase>]
> type OpenTextMode =
>     | OpenTextModeUpdating of OpenTextModeUpdating
>     | OpenTextModeWriting of OpenTextModeWriting
>     | OpenTextModeReading of OpenTextModeReading
> ```
> 
> then you can do:
> 
> ```fsharp
>     let result = builtins.``open``(StringPath "test.txt", OpenTextModeWriting OpenTextModeWriting.Write)
> ```
> 
> But it feels a bit verbose, so I think it's OK to collapse them as you suggest. The inspiration for the code is btw:
> 
> * https://github.com/python/typeshed/blob/master/stdlib/_typeshed/__init__.pyi#L156
> * https://github.com/python/typeshed/blob/master/stdlib/io.pyi#L165

It might be an error I induced, but that was the first thing I tried and the compilation wasn't quite right. Are we supporting nested union structures? I noticed that a lot of the class features in F# aren't being translated, so I thought it might be a wip

**Comment by dbrattli** [comment]
Closing because of #58

**Pull request closed without merging**